WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingCategoriesFragment.kt

```diff
@@ -0,0 +1,239 @@
+        actionListener = null
+    }
+
+    override fun onStart() {
+        super.onStart()
+        dispatcher.register(this)
+    }
+
+    override fun onStop() {
+        dispatcher.unregister(this)
+        super.onStop()
+    }
+
+    override fun onResume() {
+        val needsRequestLayout = requireArguments().getBoolean(NEEDS_REQUEST_LAYOUT)
+        if (needsRequestLayout) {
+            requireActivity().window.decorView.requestLayout()
+        }
+        super.onResume()
+    }
+
+    companion object {
+        const val TAG = "prepublishing_categories_fragment_tag"
+        const val NEEDS_REQUEST_LAYOUT = "prepublishing_categories_fragment_needs_request_layout"
+        @JvmStatic fun newInstance(
+            site: SiteModel,
+            needsRequestLayout: Boolean
+        ): PrepublishingCategoriesFragment {
+            val bundle = Bundle().apply {
+                putSerializable(WordPress.SITE, site)
+                putBoolean(NEEDS_REQUEST_LAYOUT, needsRequestLayout)
+            }
+            return PrepublishingCategoriesFragment().apply { arguments = bundle }
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        initBackButton()
+        initAddCategoryButton()
+        initRecyclerView()
+        initSwipeToRefreshHelper()
+        initViewModel()
+        super.onViewCreated(view, savedInstanceState)
+    }
+
+    private fun initBackButton() {
+        back_button.setOnClickListener {
+            viewModel.updateCategories()
+            viewModel.onBackButtonClicked()
+        }
+    }
+
+    private fun initAddCategoryButton() {
+        add_new_category.setOnClickListener {
+            viewModel.onAddNewCategoryClicked()
+        }
+    }
+
+    private fun initRecyclerView() {
+        recycler_view.layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
+        recycler_view.adapter = PrepublishingCategoriesAdapter(
+                onCheckChangeListener = { id, checked ->
+                    if (checked) {
+                        viewModel.addSelectedCategory(id)
+                    } else {
+                        viewModel.removeSelectedCategory(id)
+                    }
+                }, context = requireContext()
+        )
+        recycler_view.addItemDecoration(
+                DividerItemDecoration(
+                        recycler_view.context,
+                        DividerItemDecoration.VERTICAL
+                )
+        )
+    }
+
+    private fun initViewModel() {
+        viewModel = ViewModelProviders.of(this, viewModelFactory)
+                .get(PrepublishingCategoriesViewModel::class.java)
+        startObserving()
+    }
+
+    private fun initSwipeToRefreshHelper() {
+        swipeToRefreshHelper = WPSwipeToRefreshHelper.buildSwipeToRefreshHelper(ptr_layout,
+                SwipeToRefreshHelper.RefreshListener {
+                    if (!NetworkUtils.checkConnection(requireContext())) {
```

Comment:
Have you considered moving this logic into the VM?

Reply:
Going to remove PTR all together because it doesn't belong in a bottom sheet. 👍

Reply:
Separate task in the issue #13066.

Reply:
in commit cd27ec0
After I removed the SwipeRefreshLayout, miraculously the scrolling of the recyclerview works like a charm. 🥳